datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -80,7 +81,7 @@ pub(crate) fn prune_row_groups_by_statistics(
             let pruning_stats = RowGroupPruningStatistics {
                 parquet_schema,
                 row_group_metadata: metadata,
-                arrow_schema: predicate.schema().as_ref(),
+                arrow_schema,
```

Comment:
This is the actual fix -- to use the file schema rather than the table schema to lookup columns